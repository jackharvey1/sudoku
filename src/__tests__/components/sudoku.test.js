import React from 'react';
import Sudoku from '../../components/sudoku';
import ShallowRenderer from 'react-test-renderer/shallow';
import { mount } from 'enzyme';
import { emptySudoku } from '../sudokus.json';

let renderer;

beforeEach(() => {
    renderer = new ShallowRenderer();
});

it('renders the Sudoku component as intended', () => {
    renderer.render(<Sudoku grid={emptySudoku} />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});

it('inputs into the correct square', () => {
    const sudoku = mount(<Sudoku grid={emptySudoku} />);
    const box = sudoku.find('Box').at(0);
    const square = sudoku.find('.Square').at(0);
    const event = { target: { value: '1' } };
    square.simulate('change', event);
    sudoku.update();
    expect(box.props().values[0]).toEqual('1');
});
