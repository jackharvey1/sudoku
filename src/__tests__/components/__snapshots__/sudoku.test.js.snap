// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Sudoku component as intended 1`] = `
<div>
  <styled.div>
    <Box
      box={0}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={1}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={2}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </styled.div>
  <styled.div>
    <Box
      box={3}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={4}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={5}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </styled.div>
  <styled.div>
    <Box
      box={6}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={7}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={8}
      lockedCells={undefined}
      onClick={undefined}
      selectedBox={undefined}
      selectedSquare={undefined}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </styled.div>
</div>
`;
